src/main/java/life/qbic/openbis/openbisclient/OpenBisClient.java

```diff
@@ -924,25 +1152,16 @@ public boolean spaceExists(String spaceCode) {
 
   @Override
   public boolean projectExists(String spaceCode, String projectCode) {
-    // TODO why do we need the space code here? Then the method should be named differently
-    ProjectSearchCriteria sc = new ProjectSearchCriteria();
-    sc.withSpace().withCode().thatEquals(spaceCode);
-    sc.withCode().thatEquals(projectCode);
-    SearchResult<Project> projects = v3.searchProjects(sessionToken, sc, new ProjectFetchOptions());
-
-    return projectCode != null && projects.getTotalCount() != 0;
+    return !v3.getProjects(sessionToken,
+        Arrays.asList(new ProjectIdentifier(spaceCode, projectCode)), new ProjectFetchOptions())
+        .isEmpty();
```

Comment:
Increases readability:
```suggestion
    List foundProjects = v3.getProjects(sessionToken,
        Arrays.asList(new ProjectIdentifier(spaceCode, projectCode)), new ProjectFetchOptions());
    Boolean projectExists = foundProjects.isEmpty() ? false : true;
    return projectExists;
```